pkg/serviceprovider/tokenfilter.go

```diff
@@ -33,3 +37,36 @@ var _ TokenFilter = (TokenFilterFunc)(nil)
 func (f TokenFilterFunc) Matches(ctx context.Context, matchable Matchable, token *api.SPIAccessToken) (bool, error) {
 	return f(ctx, matchable, token)
 }
+
+// MatchAllTokenFilter is a TokenFilter that match any token
+var MatchAllTokenFilter TokenFilter = TokenFilterFunc(func(ctx context.Context, binding Matchable, token *api.SPIAccessToken) (bool, error) {
+	debugLog := log.FromContext(ctx).V(logs.DebugLevel)
+	debugLog.Info("Unconditional token match", "token", token)
+	return true, nil
+})
+
+// FallBackTokenFilter is a TokenFilter that filters token with MainTokenFilter if Condition returns true or FallBackFilter if
+// Condition returns false.
+type FallBackTokenFilter struct {
```

Comment:
Don't muck like `Fallback` inside `Fallback`... What if we call outer filter `ConditionalTokenFilter` ?